orttraining/orttraining/python/orttraining_python_module.cc

```diff
@@ -270,7 +270,15 @@ std::unique_ptr<IExecutionProvider> CreateTrainingEP(
   const SessionOptions& session_options,
   const std::string& provider_type,
   const ProviderOptionsMap& provider_options_map){
-  return CreateExecutionProviderInstance(session_options, provider_type, provider_options_map);
+  auto provider = CreateExecutionProviderInstance(session_options, provider_type, provider_options_map);
+  // The CPU provider instance created by the factory doesn't create allocators by default. The session registers
+  // the allocators to allow sharing. However, in some training scenarios (particularly eager mode), no session is
+  // created but it (eager mode) relies on the allocator in the CPU provider. Hence the change below.
```

Comment:
"change":

It may introduce confusion when readers see "change" since they can't compare the current code with the history without consulting git.

Maybe something along the lines of "so we call RegisterAllocator`?